tests/main/snapd-snap/task.yaml

```diff
@@ -1,29 +1,134 @@
 summary: Ensure snapd builds as a snap
 
-# This is what the snapcraft builder on launchpad uses.
-systems: [ubuntu-16.04-64]
+# TODO: we don't need to build the snapd snap in this test on as many systems
+# when we have a way to dynamically download the snapd snap that was built for a
+# particular PR from github actions, in which case we can just download that and
+# install it here to test the things that we need to test about the snapd snap
+
+# effectively we can only build the snap using LXD + snapcraft on ubuntu systems
+# due to various bugs with other distros and snapcraft and/or LXD
+systems:
+    # snapcraft is a classic snap, can't run on ubuntu-core
+    - -ubuntu-core*
+    # LXD doesn't work properly on amazon linux, fedora, tumbleweed, or debian
+    # to build the snap, it complains about fuse.snapfuse getting EPERM when
+    # snaps are installed inside the LXD container that snapcraft creates
+    - -amazon-linux*
+    - -debian-*
+    - -fedora-*
+    - -opensuse-*
+    # LXD is unsupported on Arch, it fails to be installed with the LXD service
+    # timing out
+    - -arch-*
```

Comment:
I had this in standup notes, but I've filed an issue with lxd now too: https://github.com/lxc/lxd/issues/8643 Maybe there's some out of tree kernel patches required for it to work.